File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = {}

        for i in range(len(nums)):
            if (target - nums[i]) not in dic:
                dic[nums[i]] = i
            else:
                return [i, dic[target - nums[i]]]

File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    hashmap := make(map[int]int)

    for i, num := range nums {
        diff := target - num
        if value, ok := hashmap[diff]; ok{
            return []int{value, i}
        }

        hashmap[num] = i
    }
    return nil;
}